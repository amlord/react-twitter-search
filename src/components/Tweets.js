import React, { Component } from 'react';
import './Tweets.css';

class Tweets extends Component {

  renderTweets() {
    return (
        <ul className="Tweets_list">
            <li>TWEET</li>
        </ul>
    );
  }

  renderNoTweets() {
    return (
        <div className="Tweets_noneFound">(no tweets found)</div>
    );
  }

  render() {
    let { tweetList } = this.props;

    return (
      <div className="Tweets">
        {(tweetList && tweetList.length) ? this.renderTweets() : this.renderNoTweets()}
      </div>
    );
  }
}

export default Tweets;
