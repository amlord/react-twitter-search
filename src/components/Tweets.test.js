import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Tweets from './Tweets';

describe('Twitter Search App', () => {
  const TWEETS = shallow(<Tweets />);

  it('renders correctly', () => {
    expect( TWEETS ).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tweets />, div);
  });

  describe('no tweets', () => {
    it('display "no tweets" message', () => {
        expect( TWEETS.find('.Tweets_noneFound').length ).toBe(1);
    });
  });

  describe('tweets', () => {
    // it('displays list of tweets', () => {
    //     expect( TWEETS.find('.Tweets__list').to.have.length(1) ).toBe(true);
    // });
  });
});