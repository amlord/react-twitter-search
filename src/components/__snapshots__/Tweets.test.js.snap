// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter Search App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tweets />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Tweets_noneFound"
>
        (no tweets found)
</div>,
      "className": "Tweets",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "(no tweets found)",
        "className": "Tweets_noneFound",
      },
      "ref": null,
      "rendered": "(no tweets found)",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Tweets_noneFound"
>
          (no tweets found)
</div>,
        "className": "Tweets",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "(no tweets found)",
          "className": "Tweets_noneFound",
        },
        "ref": null,
        "rendered": "(no tweets found)",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
